modules/spring-config/src/main/java/org/mule/runtime/config/internal/context/lazy/LazyMuleArtifactContext.java

```diff
@@ -211,8 +211,49 @@ protected void registerErrors(ArtifactAst artifactAst) {
   @Override
   protected void prepareBeanFactory(ConfigurableListableBeanFactory beanFactory) {
     super.prepareBeanFactory(beanFactory);
-    trackingPostProcessor = new TrackingPostProcessor();
-    addBeanPostProcessors(beanFactory, trackingPostProcessor);
+    currentComponentInitializationState.registerTrackingPostProcessor(beanFactory);
+  }
+
+  @Override
+  public Object getBean(String name) throws BeansException {
```

Comment:
You are right, there might be a couple of good candidates like `getBean(String name, Class<T> requiredType)` and `containsBean(String name)`, etc.
There will be some for which we don't have the necessary support, for example `getBean(Class<T> requiredType)` or `getBean(String name, Object... args)`, for those cases it will remain failing as it is right now.
I'll try and let you know.

Reply:
As discussed via meet, I tried to make a best effort approach to cover the methods of `LazyMuleArtifactContext` that implement the `BeanFactory` interface.
For the ones for which we cannot support a lazy initialization attempt right now, we will be throwing `UnsupportedOperationException`.
However, we will only be throwing that after the regular `initializeComponent`/s methods are called. This is because during a regular lazy/selective initialization there were some usages of these methods, for example: [ArtifactAstConfigurationBuilder.java#L234](https://github.com/mulesoft/mule/blob/4f57b670404ec0b1c97502b8916294c90b19a38b/modules/spring-config/src/main/java/org/mule/runtime/config/internal/ArtifactAstConfigurationBuilder.java#L234).
We are assuming we don't break backwards by doing this, because it should be only in the runtime code that we have access to the `MuleArtifactContext` directly.
Custom code can, at most, inject the `Registry` and do lookups from there. In the particular case of `Registry#lookupByType` we use `ListableBeanFactory#getBeansOfType` so it shouldn't be a problem.